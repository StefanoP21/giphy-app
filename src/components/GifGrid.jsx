import { useEffect, useState } from 'react';
import { GridItem } from './GridItem';
import { getGifs } from '../helpers/getGifs';

export const GifGrid = ({ category }) => {
  const [images, setImages] = useState([]);

  const getImages = async () => {
    const newImages = await getGifs(category);
    setImages(newImages);
  };

  useEffect(() => {
    getImages();
  }, []);

  return (
    <>
      <h3>{category}</h3>
      <div className="card-grid">
        {images.map(({ id, title, url }) => (
          <GridItem key={id} title={title} url={url} />
        ))}
      </div>
    </>
  );
};
