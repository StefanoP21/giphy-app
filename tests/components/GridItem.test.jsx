import { render, screen } from '@testing-library/react';
import { GridItem } from '../../src/components/GridItem';

describe('Pruebas en <GridItem />', () => {
  const title = 'Dark Souls';
  const url = 'https://dark-souls.com/dark-souls.jpg';

  test('debe de hacer match con el snapshot', () => {
    const { container } = render(<GridItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test('debe de mostrar la imagen con el URL y el ALT indicado', () => {
    render(<GridItem title={title} url={url} />);
    // screen.debug();
    // expect(screen.getByRole('img').src).toBe(url);
    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('debe de mostrar el tÃ­tulo en el componente', () => {
    render(<GridItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
