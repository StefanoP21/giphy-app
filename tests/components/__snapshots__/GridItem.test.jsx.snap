// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GridItem /> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dark Souls"
      src="https://dark-souls.com/dark-souls.jpg"
    />
    <p>
      Dark Souls
    </p>
  </div>
</div>
`;
